hypothesis_bio/pdb.py

```diff
@@ -0,0 +1,287 @@
+# -*- coding: utf-8 -*-
+
+"""Main module."""
+
+from datetime import date
+from string import ascii_letters, digits
+
+from hypothesis.strategies import (
+    booleans,
+    characters,
+    composite,
+    dates,
+    floats,
+    integers,
+    sampled_from,
+    text,
+)
+
+ACHAR = ascii_letters
+ATOM = "AUCGTNWSMKRYBDHV"
+ALPHANUMERIC = ACHAR + digits
+
+
+@composite
+def generate_date(draw):
+    """Generates a value of type Date in PDB format
+    """
+    sampled_date = draw(dates(min_value=date(1901, 1, 1), max_value=date.today()))
+    month = sampled_date.strftime("%b").upper()
+    year = sampled_date.strftime("%Y")[2:]
+    day = sampled_date.strftime("%d")
+    day = day.rjust(2, "0")
+    return day + "-" + month + "-" + year
+
+
+@composite
+def generate_idcode(draw):
+    """Generates a value of type IDCode in PDB format
+    """
+    first_char = str(draw(integers(min_value=0, max_value=9)))
```

Comment:
Hypothesis' [`from_regex` strategy](https://hypothesis.readthedocs.io/en/latest/data.html#hypothesis.strategies.from_regex) would be best here

this would become

```py
return draw(from_regex(r"[0-9][a-zA-Z0-9]{3}", fullmatch=True))
```